### PR TITLE
Cache viewed notifications in local storage

### PR DESCRIPTION
**Pull request by ericljpemberton**
Stop users from getting bombarded with messages each time they use the app...

I guess there isn't an easy way for this to work devices? i.e. at the account level?

### Test

Would like to test with more than one notification... Anyone have the POST handy for notifications?

### CODE REVIEW

**Comment by ericljpemberton** [comment]
@schneider-clarisse take a look when you have a minute?

File: src/js/stores/NotificationStore.jsx [hunk 1]
```diff
@@ -66,6 +77,15 @@ class NotificationStore extends EventEmitter {
     self.notifications = self.notifications.filter((notification) => {
       return notification.id != notificationParam.id;
     });
+
+    // Set this notification as viewed so it doesn't appear again
+    let viewedNotifications = JSON.parse(localStorage.getItem("viewedNotifications"));
```
Comment (schneider-clarisse) [inline src/js/stores/NotificationStore.jsx:81]: do you need to parse it twice?

**Review by schneider-clarisse — CHANGES_REQUESTED** [review]
lgtm other than double parsing

File: src/js/stores/NotificationStore.jsx (hunk 1 above)
Comment (ericljpemberton) [reply src/js/stores/NotificationStore.jsx:81]: @schneider-clarisse I missed removing the parse from the constructor which is my mistake. We do parse twice in the code. The reasoning I had behind this was that every time we read or want to append to it we should make sure they have the newest version of this list. (They may have it open in multiple chrome windows and keep updating it from these different windows. If that's the case we want to be able to reflect these changes across all windows). 

Although I guess this still wouldn't cover all instances :confused: . I can move it back to the constructor if you think that's better.

Comment (schneider-clarisse) [reply src/js/stores/NotificationStore.jsx:81]: nah that's fine just wanted to double check that there was a true reason

**Review by schneider-clarisse — APPROVED** [review]

**Review by larajanecka — APPROVED** [review]

**Pull request merged**